src/dcp_thread.c

```diff
@@ -167,10 +167,16 @@ static void *dcp_tx_thread_routine(void *data)
 	unsigned char source[ETH_ALEN];
 	uint64_t sequence_counter = 0;
 	unsigned int if_index;
-	int socket_fd;
+	int ret, socket_fd;
 
 	socket_fd = thread_context->socket_fd;
 
+	ret = get_interface_mac_address(app_config.dcp_interface, source, ETH_ALEN);
```

Comment:
Great catch! However, no need to query the source mac address again. It is already saved in `thread_context->source`.

This should solve the problem a few lines below:

`dcp_initialize_frames(thread_context->tx_frame_data, app_config.dcp_num_frames_per_cycle, thread_context->source);`

Reply:
Yes, that is the better way and works.